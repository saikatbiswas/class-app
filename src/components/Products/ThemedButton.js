import React, {useContext} from 'react';

const ThemedButton = (props) => {
    const theme = useContext(props.context);
    console.log('theme', theme)
    return (
        // <div>hfghfg</div>
      <button style={{ background: theme.background, color: theme.foreground, fontSize: theme.fontSize }}>
        I am styled by theme context!
      </button>
    );
};

export default ThemedButton; 