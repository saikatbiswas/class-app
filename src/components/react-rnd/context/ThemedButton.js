import React, { Component } from "react";
import { ThemeContext } from './ContextRnd';

class ThemedButton extends Component {
  // Assign a contextType to read the current theme context.
  // React will find the closest theme Provider above and use its value.
  // In this example, the current theme is "dark".
  // static contextType = ThemeContext;

  constructor(props) {
    super(props);
    this.state = { counter: 0 };
    this.handleClick = this.handleClick.bind(this);
  }
  
  handleClick() {
    this.setState(({counter}) => ({
      counter: counter + 1
    }));
  }

  render() {
    console.log('sdfsdf',this.context);
    if (this.state.counter === 5) {
      // Simulate a JS error
      throw new Error('I crashed!');
    }
    return (
      // <button style={{ background: this.context.background, color: this.context.foreground, fontSize: this.context.fontSize }}>
      //   I am styled by theme context!
      // </button>
      <button onClick={this.handleClick}>
        I am styled by theme context! {this.state.counter}
      </button>
    )
  }
}



// import React, {useContext} from 'react';

// const ThemedButton = (props) => {
//     // const theme = useContext();
//     // console.log('theme', theme)
//     return (
//         <div>hfghfg</div>
//       // <button style={{ background: theme.background, color: theme.foreground, fontSize: theme.fontSize }}>
//       //   I am styled by theme context!
//       // </button>
//     );
// };

export default ThemedButton; 