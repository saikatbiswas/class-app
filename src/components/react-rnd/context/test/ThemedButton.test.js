import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ThemedButton from "../ThemedButton";
import { findByTestAttr } from "../../../../../test/testUtils"; 


const setup = ()=>{
    return shallow(<ThemedButton />);
}

test('Render card block without error', ()=>{
    const wrapper = setup()
    const component = findByTestAttr(wrapper, 'context-button');
    // expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
});