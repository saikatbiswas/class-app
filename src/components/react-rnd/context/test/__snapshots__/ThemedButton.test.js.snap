// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render card block without error 1`] = `
<div
  className="context-button2"
  data-test="context-button"
>
  <button
    onClick={[Function]}
  >
    I am styled by theme context! 
    0
  </button>
</div>
`;
